src/XamlX/Helpers/DeterministicName.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System.Text;
+using System.Linq;
+using System.Threading;
+using System.Security.Cryptography;
+using System;
+
+namespace XamlX.Helpers
+{
+    public class DeterministicName
```

Comment:
MSBuild task assembly is reused across builds, so your counter will stay in memory.